Realm/ObjectServerTests/SwiftUIServerTests.swift

```diff
@@ -0,0 +1,358 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import RealmSwift
+import XCTest
+import SwiftUI
+import Combine
+
+#if canImport(RealmTestSupport)
+import RealmSwiftSyncTestSupport
+import RealmSyncTestSupport
+#endif
+
+@available(OSX 11, *)
+class SwiftUIServerTests: SwiftSyncTestCase {
+    override func setUp() {
+        super.setUp()
+        let config = Realm.Configuration(objectTypes: [SwiftHugeSyncObject.self])
+        Realm.Configuration.defaultConfiguration = config
+    }
+
+    override func tearDown() {
+        cancellables.forEach { $0.cancel() }
+        cancellables = []
+        super.tearDown()
+    }
+
+    var cancellables: Set<AnyCancellable> = []
+
+    // MARK: - AsyncOpen
+    func testAsyncOpenOpenRealm() throws {
+        _ = try logInUser(for: basicCredentials())
+
+        let ex = expectation(description: "download-realm-async-open")
+        let asyncOpen = AsyncOpen(appId: appId, partitionValue: #function)
+        asyncOpen.projectedValue
+            .sink { asyncOpenState in
+                if case let .open(realm) = asyncOpenState {
+                    XCTAssertNotNil(realm)
+                    ex.fulfill()
+                }
+            }
+            .store(in: &cancellables)
+        waitForExpectations(timeout: 10.0)
+        asyncOpen.cancel()
```

Comment:
I think the bulk of these tests could be extracted to a helper function pretty easily. Something like `func asyncOpen(partitionValue: String, handler: (AsyncOpenState, XCTestExpectation) -> Void)` which contains these lines and then calls the provided callback inside the `.sink()` function would reduce a bunch of these tests to just the code that's different between them.